packages/preview-middleware-client/src/utils/version.ts

```diff
@@ -0,0 +1,15 @@
+import type { SingleVersionInfo } from '../../types/global';
+import VersionInfo from 'sap/ui/VersionInfo';
+import Log from 'sap/base/Log';
+
+/**
+ * Retrieve the UI5 version from sap.ui.core library
+ */
+export async function getUi5Version() {
+    let version = ((await VersionInfo.load({ library: 'sap.ui.core' })) as SingleVersionInfo)?.version;
+    if (!version) {
+        Log.error('Could not get UI5 version of application. Using 1.121.0 as fallback.');
+        version = '1.121.0';
```

Comment:
Why do we set this particular version as fallback? We need to support versions from `1.71` and there are quite a few things that will not work if we assume it is `1.121`. If we can not determine the correct version then I suspect it will be quite likely that the following code will not work correctly, so I'm not convinced that it is worth continuing in such case.

Reply:
@tobiasqueck suggested to use that version as fallback instead of throwing an error. See [resolved discussion above](https://github.com/SAP/open-ux-tools/pull/2208#discussion_r1699871477)

Reply:
ok, lets keep it likes this for now. But I think we should have another look at this later, perhaps when trying to unify the version handling logic. Regular preview can perhaps function without knowing exact version, but then it shouldn't use this API and use some other heuristics to check if the required APIs are there from UI5. 

But for adaptation editor it is more critical and without knowing UI5 version there I don't think it makes sense to continue. It is too likely that either the editor will crash anyways in a less graceful way or the editor will not work correctly.